dmutils/formats.py

```diff
@@ -22,14 +23,18 @@ def dateformat(value, default_value=None):
     return _format_date(value, default_value, DISPLAY_DATE_FORMAT, localize=False)
 
 
-def datetimeformat(value, default_value=None, localize=True):
+def datetimeformat(value, default_value=None, localize=True, format_type=DISPLAY_DATETIME_FORMAT):
     # en_GB locale uses uppercase AM/PM which contravenes our style guide
-    formatted_date = _format_date(value, default_value, DISPLAY_DATETIME_FORMAT, localize=localize)
+    formatted_date = _format_date(value, default_value, format_type, localize=localize)
     if formatted_date:
         return formatted_date.replace('AM', 'am').replace('PM', 'pm').replace(" 0", " ")
     return formatted_date
 
 
+def localizeddatetimeformat(value, default_value=None, localize=True):
```

Comment:
This is all fine, we end up with questions of naming things, which isn't very interesting so feel free to ignore, but here goes anyway...

ISTM that the existing function is already supposed to be a "localized" function, at least by default, seeing as it has that parameter `localize=True` by default.

I'm not sure defaulting to the existing behaviour is right, and we've ended up in a position where _by default_ dates come out in UTC, but everything else comes out in London time - but depending on which function you use, it might not have a time zone.

None of this is going to breed any kind of consistency in our applications. Potentially we want two things:

1) A way to get UTC date-only thing - this function should be named in a particular way so that nobody uses it by mistake. We can bump versions and go backward-incompatible if necessary, I think.

2) A way to get a normal date-only string that converts to London TZ time by default.

3) A way to get a full date-time, that also converts to London time by default, and always includes the time zone indicator.

Reply:
I guess this is a symptom of the wider problem of 'date formats in the Digital Marketplace are  already wildly inconsistent'. For example, we import the raw `DATE_FORMAT` and `DISPLAY_DATE_FORMAT` (etc) strings in a bunch of places (content loader, scripts, api validation) without passing them through these `localize` functions in the utils. And in a bunch of other places we don't use the utils patterns at all! Some of these dates are displayed to users (eg in script generated emails), some are for filenames, it's a bit of a mess. I sense a tech debt ticket coming on 😸 

However, in general I agree that **localised should be the default for displaying dates to users** unless it's the 23:59:59 deadline case (where daylight savings could tip it over into the next day, causing confusion for people applying). 

Just to sanity check - the third case you list is what we've just implemented in this PR (`Monday 26 March 2012 at 12:59am BST`)?

Reply:
Sanity check - yes (my complaint there is just the name is not the obvious one you would choose, if you didn't look carefully at the code).

Reply:
Fair enough. I think we went for `localiseddateformat` as the counterpart to `utcdateformat`, but we can rename to something like... `datetimezoneformat` (ugh, must be something better!)